File: introduction-to-computing/seatwork_lab_exercises/atm_machine.py
global MY_SAVINGS
MY_NAME = "John Christian B. Dacasin"
MY_SAVINGS = 50000
MY_ACCOUNT_NUMBER = 123456789
MY_PIN_NUMBER = 123456

account_number = int(input("Please enter Account Number: "))
pin_number = int(input("Please enter PIN Number: "))
print("==================================================")

while(not(MY_ACCOUNT_NUMBER == account_number and MY_PIN_NUMBER == pin_number)):
    print("Incorrect account details, please try again.")
    account_number = int(input("Please enter Account Number: "))
    pin_number = int(input("Please enter PIN Number: "))
    print("==================================================")

print(f"Welcome, {MY_NAME}!")
print("==================================================")

def balance():
    print(f"Your current balance is ₱{MY_SAVINGS}")
    print("==================================================")

def withdraw():
    global MY_SAVINGS
    withdraw_amount = float(input("Please enter withdrawal amount (₱): "))
    print("==================================================")
    while(withdraw_amount > MY_SAVINGS or withdraw_amount < 0):
        print("Invalid withdrawal amount, please enter valid amount.")
        withdraw_amount = float(input("Please enter withdrawal amount: "))
        print("==================================================")
    MY_SAVINGS = MY_SAVINGS - withdraw_amount
    print(f"You have succesfully withdrawn ₱{withdraw_amount}")
    print(f"Your new current balance is ₱{MY_SAVINGS}")
    print("==================================================")

def deposit():
    global MY_SAVINGS
    deposit_amount = float(input("Please enter deposit amount (₱): "))
    print("==================================================")
    while(deposit_amount < 0):
        print("Invalid deposit amount, please enter valid amount.")
        deposit_amount = float(input("Please enter deposit amount: "))
        print("==================================================")
    MY_SAVINGS = MY_SAVINGS + deposit_amount
    print(f"You have succesfully deposited ₱{deposit_amount}")
    print(f"Your new current balance is ₱{MY_SAVINGS}")
    print("==================================================")

choice = None
while(not choice == 0):
    print("0 - Exit\n1 - Balance\n2 - Withdraw\n3 - Deposit")
    print("==================================================")
    choice = int(input("Please enter from the list: "))
    print("==================================================")
    if choice == 0:
        break
    elif choice == 1:
        balance()
    elif choice == 2:
        withdraw()
    elif choice == 3:
        deposit()

File: fundamentals-of-programming/week_11/ATMMachine.java
import java.util.Scanner;

public class ATMMachine {
    private static float initialSavings = 0;
    private static final long myAccountNumber = 123456789L;
    private static final int myPinNumber = 123456;
    private static final Scanner input = new Scanner(System.in);

    private static byte getInput(Scanner input) {
        System.out.println("1 - Check Balance\n2 - Withdrawal\n3 - Deposit");
        System.out.println("--------------------------");
        System.out.print("Please enter your choice: ");
        byte choice = input.nextByte();
        while(choice > 3 && choice < 0) {
            System.out.println("--------------------------");
            System.out.print("Invalid choice, please enter correct choice: ");
            choice = input.nextByte();
        }
        System.out.println("--------------------------");
        return choice;
    }
    private static void checkBalance() {
        System.out.println("Your balance is " + initialSavings);
        System.out.println("--------------------------");
    }

    private static void withdrawal() {
        System.out.print("Enter amount to withdraw: ");
        float withdrawAmount = input.nextFloat();
        while(withdrawAmount > initialSavings && withdrawAmount < 0) {
            System.out.println("--------------------------");
            System.out.print("Invalid withdrawal amount. Enter correct amount: ");
            System.out.println("--------------------------");
            withdrawAmount = input.nextFloat();
        }
        initialSavings = initialSavings - withdrawAmount;
        checkBalance();
    }

    private static void deposit() {
        System.out.print("Enter amount to deposit: ");
        float depositAmount = input.nextFloat();
        while(depositAmount < 0) {
            System.out.println("--------------------------");
            System.out.print("Invalid deposit amount. Enter correct amount: ");
            System.out.println("--------------------------");
            depositAmount = input.nextFloat();
        }
        initialSavings = initialSavings + depositAmount;
        checkBalance();
    }

    public static void main(String[] args) {
        long accountNumber = 0, pinNumber = 0;
        while(!(accountNumber == myAccountNumber && pinNumber == myPinNumber)) {
            System.out.print("Enter your Account Number: ");
            accountNumber = input.nextLong();
            System.out.print("Enter your PIN Number: ");
            pinNumber = input.nextInt();
            System.out.println("--------------------------");
            if (!(accountNumber == myAccountNumber) || !(pinNumber == myPinNumber)) {
                System.out.println("Incorrect account details, please try again.");
                System.out.println("--------------------------");
            }
        }
        
        byte choice = 1;
        while(!(choice == 0)) {
            choice = getInput(input);
            switch (choice) {
                case 1 -> checkBalance();
                case 2 -> withdrawal();
                case 3 -> deposit();
            }
            System.out.print("Press any number to continue (0 to exit): ");
            choice = input.nextByte();
            System.out.println("--------------------------");
        }
        input.close();
    }    
}
